### PR TITLE
Pagination examples

### PR DESCRIPTION
**Pull request by ed-**
Tests to make sure pagination goes the way we planned it to.
Act now and I'll throw in four new bonus example query files!

### CODE REVIEW

File: tests/integration/tests/api/databases.py [hunk 1]
```diff
@@ -127,3 +127,21 @@ def test_invalid_database_name(self):
...
+        databases.append({"name": "Cogs"})
+        databases.append({"name": "Widgets"})
+
+        self.dbaas.databases.create(instance_info.id, databases)
+        if not FAKE:
+            time.sleep(5)
+        limit = 2
+        databases = self.dbaas.databases.list(instance_info.id, limit=limit)
+        print databases
+        print databases.next
```
Comment (TimSimpsonR) [inline tests/integration/tests/api/databases.py]: NGUTPR
Python 2 only style print?

**Comment by TimSimpsonR** [comment]
+1
A+++ would buy again.

File: tests/integration/tests/api/databases.py (hunk 1 above)
Comment (ed-) [inline tests/integration/tests/api/databases.py]: Well, those are only supposed to show up if the test bombs anyway, and the test isn't bombing.
Really, I don't need those anymore anyway.

**Comment by TimSimpsonR** [comment]
Oh wait, sorry, +1 rescinded temporarily.
Could you add some asserts to the pagination tests for when the special rich client list .next field _is_ None?

File: apidocs/src/resources/samples/db-instances-paged-index-response.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                }
+            ],
+            "name": "example_rack_instance",
+            "status": "ACTIVE"
+        }
+    ],
+    "links": [
+        {
+            "rel": "next",
+            "href": "https://ord.databases.api.rackspacecloud.com/instances?marker=e0846e3b-3df4-398f-1e9a-d53650573dd1&limit=5"
```
Comment (cp16net) [inline apidocs/src/resources/samples/db-instances-paged-index-response.json]: this link should have the version and account here i believe unless there is some magic happening else where.

File: apidocs/src/resources/samples/db-instances-paged-index-response.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    </instance>
+    <instance id="e0846e3b-3df4-398f-1e9a-d53650573dd1" name="example_rack_instance" status="ACTIVE">
+        <links>
+            <link href="https://ord.databases.api.rackspacecloud.com/v1.0/1234/instances/e0846e3b-3df4-398f-1e9a-d53650573dd1" rel="self"/>
+            <link href="https://ord.databases.api.rackspacecloud.com/instances/e0846e3b-3df4-398f-1e9a-d53650573dd1" rel="bookmark"/>
+        </links>
+    </instance>
+</instances>
+<links>
+    <link href="https://ord.databases.api.rackspacecloud.com/instances?marker=e0846e3b-3df4-398f-1e9a-d53650573dd1&limit=5" rel="next" />
```
Comment (cp16net) [inline apidocs/src/resources/samples/db-instances-paged-index-response.xml]: same here needs the version/acct#

**Comment by imsplitbit** [comment]
squash plz

**Comment by cp16net** [comment]
+1

**Pull request merged**